File: app/src/main/java/com/stevenfrew/beatprompter/event/PauseEvent.java
package com.stevenfrew.beatprompter.event;

public class PauseEvent extends BaseEvent
{
    public int mBeats;
    public int mBeat;
    public PauseEvent(long eventTime,int beats, int beat)
    {
        super(eventTime);
        mBeats=beats;
        mBeat=beat;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/song/event/PauseEvent.kt
package com.stevenfrew.beatprompter.song.event

/**
 * A PauseEvent tells the event processor to wait for a certain amount of time, showing a
 * gauge onscreen for the duration of the pause.
 */
class PauseEvent(
	eventTime: Long,
	val beats: Int,
	val beat: Int
) : BaseEvent(eventTime) {
	override fun offset(nanoseconds: Long): BaseEvent =
		PauseEvent(eventTime + nanoseconds, beats, beat)
}